trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/rest/TestPresigningHeadersWithRewrite.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.server.rest;
+
+import com.google.inject.Inject;
+import io.trino.aws.proxy.server.testing.RequestRewriteUtil;
+import io.trino.aws.proxy.server.testing.TestingS3PresignController;
+import io.trino.aws.proxy.server.testing.TestingS3RequestRewriter;
+import io.trino.aws.proxy.server.testing.TestingS3SecurityController;
+import io.trino.aws.proxy.server.testing.containers.S3Container.ForS3Container;
+import io.trino.aws.proxy.server.testing.harness.TrinoAwsProxyTest;
+import io.trino.aws.proxy.server.testing.harness.TrinoAwsProxyTestCommonModules.WithConfiguredBuckets;
+import io.trino.aws.proxy.spi.rest.S3RequestRewriter;
+import software.amazon.awssdk.services.s3.S3Client;
+
+@TrinoAwsProxyTest(filters = {WithConfiguredBuckets.class, RequestRewriteUtil.Filter.class})
+public class TestPresigningHeadersWithRewrite
+        extends AbstractTestPresigningHeaders
+{
+    @Inject
+    public TestPresigningHeadersWithRewrite(@ForS3Container S3Client storageClient, S3Client internalClient, TestingS3PresignController presignController, TestingS3SecurityController securityController, S3RequestRewriter requestRewriter)
```

Comment:
Same as before, this can request `TestingS3RequestRewriter` if we make a concrete class